tensorboard/components/tf_tensorboard/tf-tensorboard.html

```diff
@@ -812,15 +813,22 @@ <h3>There’s no dashboard by the name of “<tt>[[_selectedDashboard]]</tt>.”
         this._lastReloadTime = new Date().toString();
       },
 
-      _fetchActiveDashboards() {
+      _computeActiveDashboard() {
+        return this._dashboardData
+            .map((d) => d.plugin)
+            .filter((dashboardName) => {
+                return this._pluginsListing[dashboardName] &&
+                    this._pluginsListing[dashboardName].enabled;
+            });
+      },
+
+      _fetchPluginsListing() {
         this._canceller.cancelAll();
-        const updateActiveDashboards = this._canceller.cancellable(result => {
+        const updatePluginsListing = this._canceller.cancellable(result => {
           if (result.cancelled) {
             return;
           }
-          const activePlugins = result.value;
-          this._activeDashboards = this._dashboardData.map(
-              d => d.plugin).filter(p => activePlugins[p]);
+          this._pluginsListing = result.value;
```

Comment:
The renames/refactors in this commit make the `tf-tensorboard` logic
easier to follow; thanks!